batch/batches/Archive/KAsyncLiveEntryArchive.class.php

```diff
@@ -0,0 +1,136 @@
+        $pushNotificationParameter->key = "entryId";
+        $pushNotificationParameter->value = $liveEntryString;
+        $pushEventNotificationParameterArray = array();
+        $pushEventNotificationParameterArray[] = $pushNotificationParameter;
+        $pushNotificationParams = new KalturaPushNotificationParams();
+        $pushNotificationParams->userParams = $pushEventNotificationParameterArray;
+
+        KBatchBase::impersonate($partnerId);
+        KBatchBase::$kClient->eventNotificationTemplate->sendCommand('POLLS_PUSH_NOTIFICATIONS',
+            $pushNotificationParams, KalturaPushNotificationCommandType::CLEAR_QUEUE);
+        KBatchBase::$kClient->eventNotificationTemplate->sendCommand('PUBLIC_QNA_NOTIFICATIONS',
+            $pushNotificationParams, KalturaPushNotificationCommandType::CLEAR_QUEUE);
+        KBatchBase::$kClient->eventNotificationTemplate->sendCommand('USER_QNA_NOTIFICATIONS',
+            $pushNotificationParams, KalturaPushNotificationCommandType::CLEAR_QUEUE);
+        KBatchBase::unimpersonate();
+    }
+
+    private static function updateEntriesData(KalturaLiveStreamEntry $liveEntry, KalturaBaseEntry $vodEntry)
+    {
+        $currentDate = date("M-d-Y H:i");
+        $updatedVodEntry = new KalturaMediaEntry();
+        $updatedVodEntry->name = $liveEntry->name. " " . $currentDate;
+        $updatedVodEntry->description = $liveEntry->description;
+        $updatedVodEntry->tags = $liveEntry->tags;
+        $updatedVodEntry->displayInSearch = 1;
+        KBatchBase::$kClient->baseEntry->update($vodEntry->id, $updatedVodEntry);
+
+        $updatedLiveEntry = new KalturaLiveStreamEntry();
+        $updatedLiveEntry->redirectEntryId = "";
+        $updatedLiveEntry->recordedEntryId = "";
+        KBatchBase::$kClient->baseEntry->update($liveEntry->id, $updatedLiveEntry);
+    }
+
+    private static function deleteCuePoints($entryId, $notDeletedCuePointTags)
+    {
+        $cuePointsCount = self::getCuePointCount($entryId);
+        $loopIterations = ceil($cuePointsCount / self::MAX_CUE_POINTS_PER_PAGE);
+        $filter = self::getCuePointFilter($entryId);
+        $pager = new KalturaFilterPager();
+        $pager->pageSize = self::MAX_CUE_POINTS_PER_PAGE;
+
+        $notDeletedCuePointTagsArray = explode(",", $notDeletedCuePointTags);
+
+        for ($i = 1; $i <= $loopIterations; $i++)
+        {
+            $pager->pageIndex = $i;
+            try
+            {
+                $cuePointsList = KBatchBase::$kClient->cuePoint->listAction($filter, $pager);
+                foreach ($cuePointsList->objects as $cuePoint)
+                {
+                    $cuePointTags = explode(',', $cuePoint->tags);
+                    $relevantTags = array_intersect($notDeletedCuePointTagsArray, $cuePointTags);
+                    if (count($relevantTags) == 0)
+                    {
+                        KBatchBase::$kClient->cuePoint->updateStatus($cuePoint->id, CuePointStatus::DELETED);

```

Comment:
The try - catch should apply only on specific cue point rather then all the process, since it is possible that in other PAI call user will delete a cuepoint from your list , but you will still need to delete the others.